### PR TITLE
feat(integration): add multiplexing support to Pinterest Tag

### PR DESCRIPTION
**Pull request by shrouti1507**
… added

## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/659)

### CODE REVIEW

File: integrations/PinterestTag/browser.js [hunk 1]
```diff
@@ -1,31 +1,28 @@
 /* eslint-disable class-methods-use-this */
-import get from "get-value";
-import logger from "../../utils/logUtil";
+import get from 'get-value';
```
Comment (utsabc) [inline integrations/PinterestTag/browser.js:2]: By default we use `""` not `'` can we please update it?

Comment (saikumarrs) [reply integrations/PinterestTag/browser.js:2]: We are now slowly moving away from using `"`. It's been active on v1.1 branches now.

Comment (utsabc) [reply integrations/PinterestTag/browser.js:2]: Ok

File: integrations/PinterestTag/browser.js [hunk 2]
```diff
@@ -132,18 +129,16 @@ export default class PinterestTag {
    * This function will generate required pinterest object to be sent.
    * getRawPayload() will generate all the destination property excepts lineItems
    * If rudder payload has products array then line_items is generated
-   * In case if the call is for event which has flag hasEmptyProducts to true, it will generate all
-   * properties including lineItems even if it does not have products array in it ex: Product Added
+   * It will generate all properties including lineItems even if it does not have products array in it ex: Product Added
```
Comment (ItsSudip) [inline integrations/PinterestTag/browser.js:132]: Isn't these two lines conflicting?

File: integrations/PinterestTag/browser.js [hunk 3]
```diff
@@ -165,42 +160,51 @@ export default class PinterestTag {
...
+      eventNames = keyMap[event];
     }
-    const destinationEvent = eventMapping.find((p) =>
-      p.src.includes(event.toLowerCase())
-    );
-    if (!destinationEvent) {
-      return {
-        dest: "custom",
-      };
+    if (isDefined(eventNames)) {
```
Comment (ItsSudip) [inline integrations/PinterestTag/browser.js:185]:
```suggestion
    if (eventNames) {
```

File: integrations/PinterestTag/browser.js [hunk 4]
```diff
@@ -165,42 +160,51 @@ export default class PinterestTag {
...
+      return eventNames;
     }
-    return destinationEvent;
+    /*
+    Step 2: To find if the particular event is amongst the list of standard
+            Rudderstack ecommerce events, used specifically for Pinterest Conversion API
+            mappings.
+    */
+    if (!eventNames) {
```
Comment (ItsSudip) [inline integrations/PinterestTag/browser.js:193]: This check is not required. We are already returning if `eventNames` is defined.

File: integrations/PinterestTag/browser.js [hunk 5]
```diff
@@ -155,52 +146,58 @@ export default class PinterestTag {
...
+        'to',
+        false,
       );
-      if (userDefinedEvent && userDefinedEvent.to) {
-        return {
-          dest: userDefinedEvent.to,
-          isUserDefinedEvent: true,
-        };
-      }
+      eventName = keyMap[event];
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply integrations/PinterestTag/browser.js]: addressed

File: integrations/PinterestTag/browser.js [hunk 6]
```diff
@@ -155,52 +146,58 @@ export default class PinterestTag {
...
     }
-    const destinationEvent = eventMapping.find((p) =>
-      p.src.includes(event.toLowerCase())
-    );
-    if (!destinationEvent) {
-      return {
-        dest: "custom",
-      };
+    if (isDefined(eventName)) {
+      return [...eventName];
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply integrations/PinterestTag/browser.js]: addressed

File: integrations/PinterestTag/browser.js (hunk 2 above)
Comment (shrouti1507) [reply integrations/PinterestTag/browser.js]: addressed

File: integrations/PinterestTag/browser.js (hunk 4 above)
Comment (shrouti1507) [reply integrations/PinterestTag/browser.js]: addressed

**Review by ItsSudip — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**